mongodbatlas/advanced_clusters.go

```diff
@@ -0,0 +1,230 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mongodbatlas
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+)
+
+const advancedClustersPath = "api/atlas/v1.5/groups/%s/clusters"
+
+// AdvancedClustersService is an interface for interfacing with the Clusters (Advanced)
+// endpoints of the MongoDB Atlas API.
+//
+// See more: https://docs-atlas-staging.mongodb.com/cloud-docs/docsworker-xlarge/DOCSP-14695/reference/api/clusters-advanced/
+type AdvancedClustersService interface {
+	List(ctx context.Context, groupID string, options *ListOptions) (*AdvancedClustersResponse, *Response, error)
+	Get(ctx context.Context, groupID, clusterName string) (*AdvancedCluster, *Response, error)
+	Create(ctx context.Context, groupID string, cluster *AdvancedCluster) (*AdvancedCluster, *Response, error)
+	Update(ctx context.Context, groupID, clusterName string, cluster *AdvancedCluster) (*AdvancedCluster, *Response, error)
+	Delete(ctx context.Context, groupID, clusterName string) (*Response, error)
+}
+
+// AdvancedClustersServiceOp handles communication with the Cluster (Advanced) related methods
+// of the MongoDB Atlas API
+type AdvancedClustersServiceOp service
+
+var _ AdvancedClustersService = &AdvancedClustersServiceOp{}
+
+// AdvancedCluster represents MongoDB cluster.
+type AdvancedCluster struct {
+	BaseCluster
+	ReplicationSpecs []*AdvancedRegionSpec `json:"replicationSpecs,omitempty"`
+	CreateDate       string                `json:"createDate,omitempty"`
+	RootCertType     string                `json:"rootCertType,omitempty"`
+}
+
+type AdvancedRegionSpec struct {
+	NumShards     int    `json:"numShards,omitempty"`
+	ID            string `json:"id,omitempty"`
+	ZoneName      string `json:"zoneName,omitempty"`
+	RegionConfigs []*AdvancedRegionConfig
+}
+
+type AdvancedRegionConfig struct {
+	AnalyticsSpecs      *Specs `json:"analyticsSpecs,omitempty"`
+	ElectableSpecs      *Specs `json:"electableSpecs,omitempty"`
+	ReadOnlySpecs       *Specs `json:"readOnlySpecs,omitempty"`
+	BackingProviderName string `json:"backingProviderName,omitempty"`
+	Priority            int    `json:"priority,omitempty"`
+	ProviderName        string `json:"providerName,omitempty"`
+	RegionName          string `json:"regionName,omitempty"`
+}
+
+type AdvancedAutoScaling struct {
+	DiskGB  *DiskGB  `json:"diskGB,omitempty"`
+	Compute *Compute `json:"compute,omitempty"`
+}
+
+type DiskGB struct {
+	Enabled *bool `json:"enabled,omitempty"`
+}
+
+type Specs struct {
+	DiskIOPS      float64 `json:"diskIOPS,omitempty"`
+	EbsVolumeType string  `json:"ebsVolumeType,omitempty"`
+	InstanceSize  string  `json:"instanceSize,omitempty"`
+	NodeCount     int     `json:"nodeCount,omitempty"`
```

Comment:
going to stop here but for this and all `float`/`int` can you please make sure they can't be zero if you use `omitempty`, if they can be zero then use pointers or remove `omitempty`

Reply:
Done! : )